utils/ansible-playbooks/deploy-monitoring/templates/ocp-monitor.j2

```diff
@@ -116,15 +173,42 @@ fi
 
 {% raw %}
 # Check that all nodes are Ready
+# Now with additional Finite State Automata support
+# One failure in a row will only log WARN to log file. Two or
+# more consecutive failures in a row will trigger ERROR alert.
 echo "$(datestamp) Check for NodeNotReady"
-if oc get nodes -o go-template="{{range .items}}{{.metadata.name}}{{print \"\t\"}}{{range .status.conditions}}{{if eq .type \"Ready\"}}{{.status}}{{end}}{{end}}{{print \"\n\"}}{{end}}" | grep -v True
```

Comment:
This line used to cause the cron to print the nodes that were not ready, we seem to be missing that now.

Reply:
Perhaps `test_if_all_nodes_are_ready` should STDOUT the failed nodes, and return 0/1 for the status?
https://stackoverflow.com/a/17337056

```
test_if_all_nodes_are_ready
RC="$?"
if [[ "${RC}" -eq "0" ]]; then
```

Reply:
I'm going to refactor things, since a long command is used and executed twice. Once to check for any entries other than a state of "True", and also for doing a line count if all is well.

It seems using "return" and anything other than 0 is being captured at a top level and kicking us out of the script, hence why I went to using "echo" to return status.

I will take the really long query command and dump it into a single global variable, since that will allow me to make the script cleaner, and also reduce the very slight possibility of a situation where running the command twice in a row will generate a different result. 

Stay tuned for the actual changes.